lib/database/sthDatabase.js

```diff
@@ -313,6 +313,60 @@ function getCollection(params, options, callback) {
         return process.nextTick(callback.bind(null, error));
     }
 
+    //Call fetchCollection after creating database connection if connection was not restored after database restarted
+    if (isConnectionAlive()) {
+        fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+    } else {
+        reconnectDB(callback, function() {
+            fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+        });
+    }
+}
+
+/**
+ * Check whether database is connected or not
+ */
+function isConnectionAlive() {
+    let dbConnected = true;
+    if (db === null || !db.serverConfig.isConnected()) {
+        dbConnected = false;
+    }
+    return dbConnected;
+}
+
+/**
+ * Reconnect to database
+ * @param {function} callback The Callback
+ * @param {function} callbackConnection The Callback
+ */
+function reconnectDB(callback, callbackConnection) {
+    connect({
+            authentication: sthConfig.DB_AUTHENTICATION,
+            dbURI: sthConfig.DB_URI,
+            replicaSet: sthConfig.REPLICA_SET,
+            database: sthDatabaseNaming.getDatabaseName(sthConfig.DEFAULT_SERVICE),
+            poolSize: sthConfig.POOL_SIZE,
+            authSource: sthConfig.DB_AUTH_SOURCE
```

Comment:
Have you ever consider add reconnectTries and  reconnectTries to these mongodb connection options?

Reply:
No, I have not consider to add reconnectTries to these mongodb connection options. As in `reconnectDB` if DB is not connected then it will raise the "Internal server error :500" which is desired. Why to add reconnectTries if we are getting the desired error immediately.